vnext/JSI/Shared/ChakraRuntime.cpp

```diff
@@ -425,12 +456,26 @@ facebook::jsi::Array ChakraRuntime::createArray(size_t length) {
 
 size_t ChakraRuntime::size(const facebook::jsi::Array &arr) {
   assert(isArray(arr));
-  return static_cast<size_t>(GetProperty(arr, "length").asNumber());
+
+  int result = ToInteger(GetProperty(GetChakraObjectRef(arr), "length"));
+
+  if (result < 0 || result > SIZE_MAX) {
+    throw facebook::jsi::JSINativeException("Invalid JS array length detected.");
+  }
```

Comment:
(nit) can we replace this with something like VerifyElseThrow(result < 0 || result > SIZE_MAX, "Invalid JS array length detected.")

Reply:
There are other places that might benefit from a similar pattern. I think we should put this on hold till when we work on ensuring correct exception propagation.

---
In reply to: [341273515](https://github.com/microsoft/react-native-windows/pull/3568#discussion_r341273515) [](ancestors = 341273515)